CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8.c

```diff
@@ -0,0 +1,134 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_depthwise_conv_s8.c
+ * Description:	 s8 version of depthwise convolution.
+ *
+ * $Date:        July 2019
+ * $Revision:    V.0.5.0
+ *
+ * Target Processor:  Cortex-M cores
+ *
+ * -------------------------------------------------------------------- */
+#include "arm_math.h"
+#include "arm_nnfunctions.h"
+#include "arm_nnsupportfunctions.h"
+
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup NNConv
+ * @{
+ */
+
+/**
+   * @brief Basic s8 depthwise convolution function.
+   *
+   * @note
+   *        - Refer header file for details.
+   *        - The optimized version to follow can have limitations on the ranges of certain inputs.
+   *
+   */
+arm_status arm_depthwise_conv_s8(const q7_t *input,
+                                 const uint16_t input_x,
+                                 const uint16_t input_y,
+                                 const uint16_t input_ch,
+                                 const q7_t *kernel,
+                                 const uint16_t output_ch,
+                                 const uint16_t ch_mult,
+                                 const uint16_t kernel_x,
+                                 const uint16_t kernel_y,
+                                 const uint16_t pad_x,
+                                 const uint16_t pad_y,
+                                 const uint16_t stride_x,
+                                 const uint16_t stride_y,
+                                 const q7_t *bias,
+                                 q7_t *output,
+                                 const int32_t *output_shift,
+                                 const int32_t *output_mult,
+                                 const uint16_t output_x,
+                                 const uint16_t output_y,
+                                 const int32_t output_offset,
+                                 const int32_t input_offset,
+                                 const int32_t output_activation_min,
+                                 const int32_t output_activation_max,
+                                 const uint16_t dilation_x,
+                                 const uint16_t dilation_y,
+                                 q15_t *buffer_a)
+{
+    int i_out = 0;
+    (void)dilation_x;
```

Comment:
So we don't support dilatation with this kernel ?

Reply:
Not as a first step as optimization with dilation support would be different(data packing for im2col) from that of the non-dilation version and it is not seen as a must have for embedded class networks. If we see a requirement for supporting it in future for embedded class networks, we can incrementally add support it.